sematic/plugins/kuberay_wrapper/standard.py

```diff
@@ -0,0 +1,340 @@
+        that use GPUs. Value should be json encoded into a string.
+    RAY_NON_GPU_NODE_SELECTOR:
+        The Kubernetes node selector that will be used for Ray nodes
+        that don't use GPUs. Value should be json encoded into a string.
+    RAY_GPU_TOLERATIONS:
+        The Kubernetes tokerations that will be used for Ray nodes
+        that use GPUs. Value should be json encoded into a string.
+    RAY_NON_GPU_TOLERATIONS:
+        The Kubernetes tokerations that will be used for Ray nodes
+        that don't use GPUs. Value should be json encoded into a string.
+    RAY_GPU_RESOURCE_REQUEST_KEY:
+        The key that will be used in the Kubernetes resource requests/
+        limits fields to indicate how many GPUs are required for Ray
+        head/worker nodes. If not specified, requests for more than one
+        GPU will be denied. Value should be json encoded into a string.
+    RAY_SUPPORTS_GPUS:
+        Whether Ray heads/workers can be assigned to Kubernetes nodes with
+        GPUs. Should be either set to "true" or "false".
+    """
+
+    RAY_GPU_NODE_SELECTOR = "RAY_GPU_NODE_SELECTOR"
+    RAY_NON_GPU_NODE_SELECTOR = "RAY_NON_GPU_NODE_SELECTOR"
+    RAY_GPU_TOLERATIONS = "RAY_GPU_TOLERATIONS"
+    RAY_NON_GPU_TOLERATIONS = "RAY_NON_GPU_TOLERATIONS"
+    RAY_GPU_RESOURCE_REQUEST_KEY = "RAY_GPU_RESOURCE_REQUEST_KEY"
+    RAY_SUPPORTS_GPUS = "RAY_SUPPORTS_GPUS"
+
+
+class _NeedsOverride:
+    """Non-encodable signal value to indicate a template value needs to be overridden."""
+
+    pass
+
+
+MIN_SUPPORTED_KUBERAY_VERSION = (0, 4, 0)
+
+
+_WORKER_GROUP_TEMPLATE: Dict[str, Any] = {
+    "replicas": _NeedsOverride,
+    "minReplicas": _NeedsOverride,
+    "maxReplicas": _NeedsOverride,
+    "groupName": _NeedsOverride,
+    "rayStartParams": {"block": "true"},
+    "template": {
+        "spec": {
+            "containers": [
+                {
+                    "name": "ray-worker",
+                    "image": _NeedsOverride,
+                    "lifecycle": {
+                        "preStop": {"exec": {"command": ["/bin/sh", "-c", "ray stop"]}}
+                    },
+                    "volumeMounts": [{"mountPath": "/tmp/ray", "name": "ray-logs"}],
+                    "resources": {
+                        "limits": _NeedsOverride,
+                        "requests": _NeedsOverride,
+                    },
+                }
+            ],
+            "initContainers": [
+                {
+                    "name": "init",
+                    "image": "busybox:1.28",
+                    "command": [
+                        "sh",
+                        "-c",
+                        "until nslookup "
+                        "$RAY_IP."
+                        "$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)"
+                        ".svc.cluster.local; "
+                        "do echo waiting for K8s Service $RAY_IP; sleep 2; done",
```

Comment:
this was actually the value used by Ray in their own examples, I'll leave it as they did since they probably went through some experimentation to determine a good value!